### PR TITLE
button: Increase styles specificity

### PR DESCRIPTION
**Pull request by dimitrisnl**
Change-type: patch

Connects-to: # 
See: <url> 
Depends-on: <url> 
Change-type: major|minor|patch 

---
##### Contributor checklist

- [ ] I have rebuilt the README with `npm run build:docs`
- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

**Comment by dimitrisnl** [comment]
@sradevski This finally fixes the issue I have with the website

**Comment by dimitrisnl** [comment]
Can't figure out why there is such a mismatch with generated snapshots & screens

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -36,22 +36,24 @@ const getHoverEffectOverride = (
...
-	white-space: nowrap;
-	svg {
-		color: inherit !important;
-		font-size: 0.925em;
-	}
-	/* These rules cause consistent styling when the button is rendered as a link */
-	display: inline-flex;
-	align-items: center;
-	justify-content: center;
+	&& {
```
Comment (sradevski) [inline src/components/Button/index.tsx]: Can you just add a comment why we are adding this?

**Comment by dimitrisnl** [comment]
@sradevski I didn't notice the screens before. Do they look good now?

**Comment by dimitrisnl** [comment]
As for your follow-up, no. I noticed the same behaviour in the `Banner` component.

**Comment by dimitrisnl** [comment]
I'm going to try with having a single ampersand in a separate commit.
One issue I noticed, is that it overrides the styles that come via className.

**Comment by dimitrisnl** [comment]
@balena-ci retest

**Comment by dimitrisnl** [comment]
@sradevski any clues on why would store-npm fail here? It really makes me wonder.

**Comment by sradevski** [comment]
@dimitrisnl if the commit hash is the same, it will fail (maybe it attempted publishing under the same commit hash twice). Making some small change and pushing again will fix the problem. Also, for local testing, I would recommend using `npm pack` instead.

**Comment by dimitrisnl** [comment]
@sradevski It fails due to a TableRow component. I pushed new commits and noticed the same behaviour.

**Comment by sradevski** [comment]
@dimitrisnl It seems the typings changed somewhere (React or Grommet). Can you override the `defaultValue` typings in the `Checkbox` component to match whatever it asks for?

**Comment by dimitrisnl** [comment]
It doesn't make sense to have `number` as a default value for a checkbox. Unless I'm reading it wrong.

**Comment by sradevski** [comment]
@dimitrisnl for a checkbox it doesn't make sense, but there is no `HTMLCheckboxElement` type, only `HTMLInputElement` unfortunately, so it applies to any kind of input.

File: src/components/Checkbox/index.tsx [hunk 2]
```diff
@@ -60,9 +60,10 @@ const Checkbox = ({ className, ...otherProps }: InternalCheckboxProps) => {
 };
 
 export interface InternalCheckboxProps
-	extends DefaultProps,
+	extends Omit<DefaultProps, 'defaultValue'>,
```
Comment (sradevski) [inline src/components/Checkbox/index.tsx]: Sorry about this, but what I think should have happened, is in `TableRow`, `checkboxAttributes` should be of `CheckboxProps` type instead, which will resolve this issue and make it more accurate.

Comment (dimitrisnl) [reply src/components/Checkbox/index.tsx]: Hm, still doesn't work. Wondering how did the last PR get through.

Comment (sradevski) [reply src/components/Checkbox/index.tsx]: @dimitrisnl You also need to update `Table.index`, as the `checkboxAttributes` comes from there.

**Comment by dimitrisnl** [comment]
Great! The single `&` does the trick. I noticed some changes in the local snapshots that might help us understand the issue further.

Will document it, update the screenshots/snapshots and push again.

**Comment by dimitrisnl** [comment]
@sradevski If you're good, I can merge.

**Comment by sradevski** [comment]
@dimitrisnl Some of the screenshots are a bit weird (check the Tags or Steps one for example), so something changed for sure. It seems it is only the `plain` button that is affected as far as I could tell. Maybe we relied on the CSS not being applied correctly until now? Let's get those fixed first, otherwise a lot of things will break.

**Comment by dimitrisnl** [comment]
I see @sradevski Thanks for spotting that, I'll address them.
In the meantime I'll update the deployed storybook version.

**Pull request merged**